src/hooks/sounds.tsx

```diff
@@ -49,19 +53,29 @@ export const CustomSoundsProvider: React.FC = ({ children }) => {
   const [soundConfigs, setSoundConfigs] = useState<SoundConfigMap>(
     initialValue.soundConfigs
   );
+  const [isCustomSoundsLoaded, setIsCustomSoundsLoaded] = useState<boolean>(
+    initialValue.isCustomSoundsLoaded
+  );
+  const { user } = useUser();
+  const userId = user?.uid;
 
   // Fetch the sound configs data on first load
   useEffect(() => {
-    fetchSoundConfigs().then((soundConfigs) => {
-      setSoundConfigs(soundConfigs);
-    });
-  }, []);
+    if (!userId) return;
+
+    fetchSoundConfigs()
+      .then((soundConfigs) => {
+        setSoundConfigs(soundConfigs);
+      })
+      .finally(() => setIsCustomSoundsLoaded(true));
+  }, [userId]);
```

Comment:
The `.finally()` here will run regardless of whether the `Promise` succeeds or fails. Is that intended?

I guess if we're blocking the app from loading based on this, then we probably need to signal to 'allow it to keep loading' even when there is an error.

Reply:
Yes, I'm using the`.finally()` statement to be sure it's not a blocker for the app.

Reply:
Makes sense